lmdeploy/pytorch/models/qwen.py

```diff
@@ -215,21 +209,27 @@ def _continuous_batching_forward(
         # Attention mask is not necessary in continuous batching
         hidden_states = inputs_embeds
 
-        # decoder layers
-        for idx, decoder_layer in enumerate(self.h):
-            past_key_value = (past_key_values[idx]
-                              if past_key_values is not None else None)
-            layer_outputs = decoder_layer(
+        context = self.context.context
+        max_kv_seq_length = context.max_kv_seq_length
+        # do not support use_dynamic_ntk
```

Comment:
why?

Reply:
for these reasons, we could do not support this feature for now:
1. dynamic_ntk only works on prefilling phase where input_len is greater than `seq_lenth` in config.json, which is 8192 for qwen7b
2. dynamic_ntk should assign different ntk_alphas for different sequences in a batch, which is complicated to implement in pytorch engine
3. Qwen2 has remove `dynamic_ntk`, which means it is not so important.
4. Evaluations results with opencompass suggests that it won't change the results too much:

Reply:
But seq_lenth for Qwen-14B is 2048
https://huggingface.co/Qwen/Qwen-14B/blob/main/config.json

Reply:
> But seq_lenth for Qwen-14B is 2048 https://huggingface.co/Qwen/Qwen-14B/blob/main/config.json

Yes. you are right. How do you use lmdeploy with Qwen-14B on pytorch engine? is dynamic ntk useful for you ? Why not use Qwen2?